relays/bin-substrate/src/cli/conversion_rate_update.rs

```diff
@@ -0,0 +1,224 @@
+	bridge: ConversionRateUpdateBridge,
+	#[structopt(flatten)]
+	source: SourceConnectionParams,
+	#[structopt(flatten)]
+	source_sign: SourceSigningParams,
+}
+
+/// Update conversion rate bridge.
+#[derive(Clone, Copy, Debug, EnumString, EnumVariantNames, PartialEq)]
+#[strum(serialize_all = "kebab_case")]
+pub enum ConversionRateUpdateBridge {
+	StatemineToStatemint,
+}
+
+impl ConversionRateUpdateBridge {
+	/// Return source chain runtime storage key that under which conversion rate (`FixedU128`) is
+	/// stored.
+	pub fn conversion_rate_storage_key_at_source(&self) -> StorageKey {
+		match *self {
+			// TODO (https://github.com/paritytech/parity-bridges-common/issues/1626): change once pallet is ready
+			Self::StatemineToStatemint => StorageKey(vec![42]),
+		}
+	}
+}
+
+/// Helper trait to ease bridge declarations.
+#[async_trait]
+trait ConversionRateUpdateLoop {
+	/// Source chain of the bridge.
+	type SourceChain: CliChain + ChainWithTransactions;
+	/// Target chain of the bridge.
+	type TargetChain: Chain;
+
+	/// Type that crafts the update-conversion-rate call at the source chain.
+	type UpdateConversionRateCallBuilder: UpdateConversionRateCallBuilder<Self::SourceChain>;
+
+	/// Run the conversion rate update loop.
+	async fn run_loop(data: ConversionRateUpdate) -> anyhow::Result<()>
+	where
+		Self::SourceChain:
+			CliChain<KeyPair = <Self::SourceChain as ChainWithTransactions>::AccountKeyPair>,
+		AccountIdOf<Self::SourceChain>: From<<AccountKeyPairOf<Self::SourceChain> as Pair>::Public>,
+	{
+		let source_client = data.source.into_client::<Self::SourceChain>().await?;
+		let source_transactions_mortality = data.source_sign.source_transactions_mortality;
+		let source_sign = data.source_sign.to_keypair::<Self::SourceChain>()?;
+
+		let source_transactions_params = substrate_relay_helper::TransactionParams {
+			signer: source_sign,
+			mortality: source_transactions_mortality,
+		};
+
+		let metrics = create_metrics::<_, Self::TargetChain>(data.bridge, source_client.clone())?;
+		metrics.spawn();
+
+		run_conversion_rate_update_loop::<
+			_,
+			Self::TargetChain,
+			Self::UpdateConversionRateCallBuilder,
+		>(
+			source_client,
+			source_transactions_params,
+			metrics.target_to_source_conversion_rate.shared_value_ref(),
+			metrics.source_to_base_conversion_rate.shared_value_ref(),
+			metrics.target_to_base_conversion_rate.shared_value_ref(),
+			crate::cli::relay_headers_and_messages::CONVERSION_RATE_ALLOWED_DIFFERENCE_RATIO,
+		);
+
+		Ok(())
+	}
+}
+
+/// Statemine -> Statemint conversion rate update loop.
+struct StatemineToStatemintLoop;
+
+impl ConversionRateUpdateLoop for StatemineToStatemintLoop {
+	type SourceChain = Statemine;
+	// this is a bit hacky - we are not going (?) to connect to the Statemint now, so we don't need
+	// those primitives and client crates. We're only using the `TargetChain` to get its `TOKEN_ID`.
+	// Statemint is the common god parachain of the Polkadot, so let's use Polkadot here
```

Comment:
```suggestion
	// Statemint is the common good parachain of the Polkadot, so let's use Polkadot here
```

Reply:
I guess, we should use better "common good" -> "system"